src/xbot/common/controls/sensors/MockXboxController.java

```diff
@@ -12,10 +15,74 @@ public MockXboxController(int port) {
     XYPair leftJoystickAxis;
     XYPair rightJoystickAxis;
     double leftTriggerAxis;
+    double x = 0;
+    double y = 0;
     double rightTriggerAxis;
+    private boolean xRightInverted = false;
+    private boolean yRightInverted = false;
+    private boolean xLeftInverted = false;
+    private boolean yLeftInverted = false;
+    
+    Map<Integer, Boolean> buttons = new HashMap<Integer, Boolean>();
+    Map<Integer, Double> rawAxis = new HashMap<Integer, Double>();
     
     boolean[] buttonValues = new boolean[10];
     
+    public void setRightX(double x) {
+        double value = x * (xRightInverted ? -1 : 1);
+        this.x = value;
+        setRawAxis(0, value);
+    }
+
+    public void setLeftX(double x) {
+        double value = x * (xLeftInverted ? -1 : 1);
```

Comment:
Let's say I have an instance of this mock class. If I keep inversion off, the value that I `set()` for the virtual value will be equal to the one I `get()`. If I then turn inversion on, _that will still be the case_. That seems potentially confusing to me, given that the general style of the mock class is that the imaginary value which you set is equivalent to the value that would be read from the hardware. I was wondering what the thought process was here.